exasol_script_languages_container_ci_setup/lib/__init__.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import shlex
+import subprocess
+import sys
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Optional
+
+import boto3
+import jinja2
+from botocore.exceptions import ClientError
+
+
+def upload_cloudformation_stack(aws_profile: str, yml: str, stack_name: str):
+    """
+    Use the AWS CLI to deploy a stack. (Note we don't use Boto3 here
+    because it looks much more complicated than the CLI variant and there is not really good documentation for it.
+    """
+    logging.info(f"Deploy stack with name='{stack_name}' for aws profile {aws_profile}")
+    with TemporaryDirectory() as yml_dir:
+        yml_file_path = Path(yml_dir) / "cloudformation.yaml"
+        with open(yml_file_path, "w") as yml_file:
+            yml_file.write(yml)
+        command = f"aws cloudformation deploy --stack-name {stack_name} " \
+                  f"--profile {aws_profile} --template-file {yml_file_path} " \
+                  f"--capabilities CAPABILITY_IAM"
+        logging.info(f"Invoking command: {command}")
+        completed_process = subprocess.run(shlex.split(command))
+        try:
+            completed_process.check_returncode()
+        except subprocess.CalledProcessError as e:
+            sys.exit(1)
```

Comment:
I would like to use boto3 here, but there are no examples how to deploy a stack, and it seems very laborious: 
1. Have to check if stack already exists
2. Then decide if to create a new one or update existing one
3. Then register a `waiter` because operations are asynchronous.

Using the CLI here is much easier. However it has the drawback that log-level = Debug does not print additional info (where the other functions print detailed info if loglevel=DEBUG) because boto3 also uses the standard logging library.

Reply:
The main issue with using the cli is, we need it as dependency. Also if boto3 is a bit more complicated, I still would suggest to use it. We can define it as dependency. The CLI is also based on boto3.